src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,532 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <mutex>
+#include <memory>
+
+#include "ignition/common/Console.hh"
+#include "ignition/common/Profiler.hh"
+#include "ignition/common/Image.hh"
+#include "ignition/sensors/RenderingEvents.hh"
+#include "ignition/sensors/SensorFactory.hh"
+
+#include "ignition/sensors/BoundingBoxCameraSensor.hh"
+#include "ignition/rendering/BoundingBoxCamera.hh"
+
+#include "ignition/transport/Node.hh"
+#include "ignition/transport/Publisher.hh"
+#include "ignition/msgs.hh"
+
+
+using namespace ignition;
+using namespace sensors;
+using namespace rendering;
+
+class ignition::sensors::BoundingBoxCameraSensorPrivate
+{
+  /// \brief SDF Sensor DOM Object
+  public: sdf::Sensor sdfSensor;
+
+  /// \brief True if Load() has been called and was successful
+  public: bool initialized = false;
+
+  /// \brief Rendering BoundingBox Camera
+  public: rendering::BoundingBoxCameraPtr boundingboxCamera {nullptr};
+
+  /// \brief Rendering RGB Camera to draw boxes on it and publish
+  /// its image (just for visualization)
+  public: rendering::CameraPtr camera {nullptr};
```

Comment:
```suggestion
  public: rendering::CameraPtr rgbCamera {nullptr};
```
nit: re-name to `rgbCamera` so that it's clear this camera isn't responsible for bounding boxes